File: backend/app/models/summary.py
from __future__ import annotations

from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict

Sentiment = Literal["productive", "tense", "casual", "neutral"]


class ActionItem(BaseModel):
    task: str
    assignee: str | None = None
    deadline: str | None = None


class Summary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    meeting_id: UUID
    overview: str
    decisions: list[str]
    action_items: list[ActionItem]
    topics: list[str]
    sentiment: Sentiment | None = None
    key_quotes: list[str]
    raw_ai_response: dict | None = None
    created_at: datetime


class SummaryCreate(BaseModel):
    overview: str
    decisions: list[str]
    action_items: list[ActionItem]
    topics: list[str]
    sentiment: Sentiment | None = None
    key_quotes: list[str]
    raw_ai_response: dict | None = None

File: frontend/src/types/summary.ts
export type Sentiment = 'productive' | 'tense' | 'casual' | 'neutral';

export interface ActionItem {
  task: string;
  assignee: string | null;
  deadline: string | null;
}

export interface Summary {
  id: string;
  meeting_id: string;
  overview: string;
  decisions: string[];
  action_items: ActionItem[];
  topics: string[];
  sentiment: Sentiment | null;
  key_quotes: string[];
  raw_ai_response: Record<string, unknown> | null;
  created_at: string;
}
